### ISSUE TITLE
Open Grant Proposal: 180Protocol

### ISSUE THREAD

**Issue by hasan-cuthbert** [filecoin-project/devgrants#451]
**Name of Project:** 180Protocol

**Proposal Category:** `app-dev`

**Proposer:** 
Hasan Cuthbert - `hasan-cuthbert`
Parth Shukla - `parthbond180`

**(Optional) Technical Sponsor:** 

Jonathan Victor: jonathan@protocol.ai 
Bill Schreckenstein: schreck@fil.org 
Stefaan Vervaet: stefaan.vervaet@protocol.ai

**Agree to open source all work on behalf of this RFP and dual-license under MIT and APACHE2 licenses?:**  "Yes"

# Project Description

Traditional businesses have struggled to assert their digital presence, and increasingly are at risk from platform companies. Platform companies are already worth $15 trillion in the US alone and are growing at an alarming rate, but business data is sensitive, and so trusted intermediaries or platforms are often needed to facilitate transactions. This leads to lost value potential and inefficiencies. 

180Protocol’s technology makes B2B collaboration easier by enabling Web3 for enterprise.180Protocol is a coordination infrastructure for secure multi party collaboration. It is a toolkit for collaboration built for the confidential cloud. We mobilize data at rest by overcoming barriers to deployment like privacy concerns and lack of incentive to create collaborative data products. 
By sharing data autonomously, businesses can increase their collective knowledge and enable better commercial discovery, leading to decreased costs and increased revenues. This ultimately filters down to broader society. We are excited by the enormous potential of Web3 for business.

## Value

180Protocol is a transformational toolkit that brings Web3 to institutions and enterprises. Accelerating the adoption of Web3 from traditional industries and institutions brings tremendous value to Web3. Businesses are already lagging behind in digital adoption. If they do not take control of their data assets then further disruption from value-extracting intermediaries is inevitable.

180Protocol seek to engage with decentralized technology wherever possible and see Filecoin/IPFS as the natural on-ledger storage implementation. The adoption of 180Protocol by could see the redirection of vast swathes of assets to the Filecoin/IPFS ecosystem. 

The value we add is reliant on developing traction, much like our partnership with the UK’s National Gallery. On-boarding is challenging in the institutional space, and so the primary risk we face is adoption. We combine a team of deep enterprise expertise with inventive mechanisms to help manage this risk.

## Deliverables

Technical:

- Fully integrating Filecoin as the storage layer of the 180Protocol platform.
- A set of constitutional templates for 180Protocol Data DAOs and other use cases able to be built on the Filecoin VM. 

Marketing:

- Announcing Filecoin as our storage layer on our website and across our media channels.
- Releasing partnership announcements along with blogs about the partnership
- Organizing joint AMAs with interested developers and community about the importance of decentralization of storage layer

#### Milestone 1: 	Introducing a regression based rewards engine that can be utilized within the Aggregator SDK

- Integrating regression model with Aggregator SDK
- Develop staking and attribution mechanisms

The purpose of this Milestone is to consolidate our foundational technology, incorporating the rewards engine into our Tech Stack. The regression of historic data attributions enables us to model the value of data contributions and is used for value attribution and staking mechanisms that are run by the rewards engine. 

[1.5 Months, 4/3 Developers, $22,000]
 

#### Milestone 2: 	Output data storage on Filecoin via Estuary API

- Estuary API POC with Corda
- Develop Corda - Filecoin connector for storage of confidential computing data outputs from R3 Conclave on Filecoin/IPFS
- Integrate connector with data aggregation flows for output data storage on Filecoin

This milestone establishes the decentralized storing of network assets. Initially the output data is stored on Filecoin/IPFS developing the Filecoin - Corda connector and updating the Aggregator SDK. Data assets are to be encrypted - and potentially sharded - for secure storage.  

[1.5-2 Months, 4/3 Developers, $24,000]
 

#### Milestone 3: 	Input data storage on Filecoin via Estuary API

- Integrate Corda - Filecoin connector for storing data provider data on Filecoin
- Develop provider data pipeline and integrate with Corda - Filecoin connector
- Test end to end scenario

The purpose of this milestone is to fully establish the decentralized storage of data assets throughout the 180Protocol network, integrating the Filecoin - Corda connector throughout the network pipeline. 

[1.5-2 Months, 4/3 Developers, $24,000]

#### Milestone 4: 	Filecoin VM integration and Deployment of Use Cases

- Data DAOs
  - Explore formation of 180Protocol Data DAO on Filecoin VM
  - 180Protocol DAOs can enable coalition to govern coalition behavior around data sharing
- Decentralized compute
  - Explore using Filecoin VM as a secure computation layer for multi-party computation

With Filecoin/IPFS storage integrated throughout the 180Protocol network, further integration with the Filecoin ecosystem is sought. 

[2 Months, 4/3 Developers, $30,000]

## Total Budget Requested

Estimated Total Budget: $100,000

Estimated Breakdown: 
- Salary: $80,000
- Bounty reserve for specialist tasks: $10,000
- Operational Costs: $10,000

The project is time and material based. Estimations are based on the average hourly rate of $100. Estimates of Milestones’ durations are between one and two months and specified in the section above based on feedback from the wider developer team. More precise estimations are possible as the project gets underway. 

## Maintenance and Upgrade Plans

As a foundational project, the durability of the platform is critical to the success of what is a long-term project. Envisaging a wide base of users, maintenance of the platform is very important to us. 

Once the outlined milestones have been completed, a detailed plan for long-term maintenance will be drafted by the team and stakeholders. Updates to the platform as per our project roadmap, will be made to stay relevant to our ever-changing needs. Wider development of traction and use cases, and further integration with decentralized technologies are the main pillars of platform upgrades. 

# Team

## Team Members

Commercial Team:
- Katie Lundie (CEO)
- George Cox (Marketing Lead)
- Hasan Cuthbert (Data Engineer)

Tech Team:
- Phil Holbrok (CEO)
- Conor Diveny (CFO)
- Parth Shukla (CTO)

## Team Member LinkedIn Profiles

- Katie Lundie: https://www.linkedin.com/in/katie-lundie-%E9%BE%8D%E6%84%B7%E6%9B%A6-b3151260/
- Phil Holbrok: https://www.linkedin.com/in/philip-holbrook/
- Conor Diveny: https://www.linkedin.com/in/conor-diviney-b6a569a1/
- Parth Shukla: https://www.linkedin.com/in/parthshukla1/
- George Cox: https://www.linkedin.com/in/george-cox-82a76021b/
- Hasan Cuthbert: https://www.linkedin.com/in/hasan-cuthbert-a2740a168/

## Team Website

Website: https://180protocol.com/

## Relevant Experience

Many of the team met at Cambridge whilst completing their MBA. This group would go on to found the FinTech startup Bond180. Bond180, a technology company providing technology solutions for fixed income markets, received early backing from R3 and the University of Cambridge and were hosted at the Outlier Ventures Basecamp. 

180Protocol has spawned from that group, widening the technology’s scope and redirecting its focus to the Web3.0 landscape. Our team benefits from its eclectic background in our capacity to innovate. With deep industry experience, we have benefited from early traction in the Financial Services and Arts & Culture industries, namely with our partnership with the National Gallery.

## Team code repositories

GitHub: https://github.com/180Protocol

**Comment by realChainLife** [comment]
Hi @hasan-cuthbert thank you for this proposal, we would like to fund the work outlined here. Please email devgrants@fil.org to discuss next steps.

**Comment by parthbond180** [comment]
Hi all, I wanted to provide an update on this grant. 180Protocol completed Milestones 2 and 3 of this grant as part of development of our data union SDK built on Corda and Conclave.

The team utilized best practices in cloud security and infra called envelope encryption to manage the onboarding of sensitive data seamlessly to Filecoin. We integrated with Estuary to onboard data onto Filecoin and provided a robust key and metadata management solution for data union users.

The team realized the potential of this product to aid in onboarding sensitive data sets to Filecoin and hence spun this encryption tooling as a separate CLI for the Filecoin ecosystem. This was done as part of another [grant 631 M1](https://github.com/filecoin-project/devgrants/issues/631) and the output of that is the [Encloud storage CLI](https://github.com/encloud-tech/encloud/) live today and is the easiest solution to onboard encrypted sensitive data to Filecoin.

Additionally, the team wanted to build Milestone 1 and 4 of this grant but decided to de prioritise these in favour of delivering the work for grant 631. We have built the theoretical frameworks for Milestone 1, a regression based rewards engine, of this grant but integrating this into our AggregatorSDK will make sense once there is a generalized decentralized compute layer that can be leveraged.

That brings us to Milestone 4 of this grant which aims to build a DataDAO like structure on the FVM to enable the sharing of and compute on sensitive data sets. FVM is still a reference implementation under active development and hence we aim to wait till this is stabilized.

In the meanwhile it is necessary to build the decentralized privacy preserving compute layer that can be integrated with FVMs actor interfaces.  In close consultation with the Bacalhau and COD teams, we conducted research on ascertaining the most suitable privacy compute technologies for a decentralized compute architecture as part of [grant 631 M2](https://github.com/filecoin-project/devgrants/issues/631). The outcome of this research was to build a TEE based compute layer on top of the Bacalhau project.

The research report for grant 631 M2 includes the detailed architecture on how to build a decentralized privacy compute layer. We are building this compute layer by collaborating with the Bacalhau team as part of [grant 1146](https://github.com/filecoin-project/devgrants/issues/1146), which we have recently submitted. Once this grant is delivered, we can come back to the remaining Milestones in this grant and leverage the decentralized compute layer to create DataDAOs on the FVM.
